src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -737,16 +737,11 @@ private static TypeWithAnnotations DelegateReturnTypeWithAnnotations(MethodSymbo
 
             LambdaSymbol createLambdaSymbol()
             {
-                return new LambdaSymbol(
-                                    Binder,
-                                    Binder.Compilation,
-                                    Binder.ContainingMemberOrLambda,
-                                    _unboundLambda,
-                                    parameterTypes,
-                                    parameterRefKinds,
-                                    parameterEffectiveScopesBuilder.ToImmutable(),
-                                    refKind: default,
-                                    returnType: default);
+                // This can be null only if `HasExplicitlyTypedParameterList == false`
+                // which has been ruled out at this point.
+                var lambdaSymbol = LambdaForParameterDefaultValues;
```

Comment:
>LambdaForParameterDefaultValues [](http://example.com/codeflow?start=35&length=31)

It looks like the comment to reuse `LambdaForParameterDefaultValues` was in response to [https://github.com/dotnet/roslyn/pull/63448#discussion_r955307515](https://github.com/dotnet/roslyn/pull/63448#discussion_r955307515).

@333fred, was it your intention that we use `LambdaForParameterDefaultValues` to populate `parameterDefaultValues` above, and for the rest of this `InferDelegateType()` method, or for some other location?

Reply:
The goal was only to avoid rebinding default parameter values when it isn't necessary.

Reply:
Thanks. I've tried implementing that, but it does not seem easily doable. The default parameter value itself could be reused, but its binding diagnostics would be lost. And they are merged with other lambda diagnostics, so copying them would result in duplicates. See [`SourceComplexParameterSymbol.DefaultSyntaxValue`](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Source/SourceComplexParameterSymbol.cs,00e705309035635c).

I've updated this branch to only remove the PROTOTYPE comment and also use `CreateLambdaSymbol` instead of `new LambdaSymbol` (4285ab32704696dd2b012aa9e31e533ab5655809). The latter is not particularly related, but it seems right and is something I did while working on this.